packages/core/src/debug/debug_node.ts

```diff
@@ -278,18 +257,75 @@ class DebugElement__POST_R3__ extends DebugNode__POST_R3__ implements DebugEleme
   }
 
   get attributes(): {[key: string]: string | null;} {
-    // https://angular-team.atlassian.net/browse/FW-719
-    throw notImplemented();
+    const attributes: {[key: string]: string | null;} = {};
+    const element = this.nativeElement;
+    if (element) {
+      const eAttrs = element.attributes;
+      for (let i = 0; i < eAttrs.length; i++) {
+        const attr = eAttrs[i];
+        attributes[attr.name] = attr.value;
+      }
+    }
+    return attributes;
   }
 
   get classes(): {[key: string]: boolean;} {
-    // https://angular-team.atlassian.net/browse/FW-719
-    throw notImplemented();
+    const classes: {[key: string]: boolean;} = {};
+    const element = this.nativeElement;
+    if (element) {
+      const lContext = loadLContextFromNode(element);
+      const lNode = lContext.lView[lContext.nodeIndex];
+      if (isStylingContext(lNode)) {
+        for (let i = StylingIndex.SingleStylesStartPosition; i < lNode.length;
+             i += StylingIndex.Size) {
+          if (isClassBased(lNode, i)) {
+            const className = getProp(lNode, i);
+            const value = getValue(lNode, i);
+            if (typeof value == 'boolean') {
+              // we want to ignore `null` since those don't overwrite the values.
+              classes[className] = value;
+            }
+          }
+        }
+      } else {
+        // Fallback, just read DOM.
+        const eClasses = element.classList;
+        for (let i = 0; i < eClasses.length; i++) {
+          classes[eClasses[i]] = true;
+        }
+      }
+    }
+    return classes;
   }
 
   get styles(): {[key: string]: string | null;} {
-    // https://angular-team.atlassian.net/browse/FW-719
-    throw notImplemented();
+    const styles: {[key: string]: string | null;} = {};
+    const element = this.nativeElement;
+    if (element) {
+      const lContext = loadLContextFromNode(element);
+      const lNode = lContext.lView[lContext.nodeIndex];
+      if (isStylingContext(lNode)) {
+        for (let i = StylingIndex.SingleStylesStartPosition; i < lNode.length;
+             i += StylingIndex.Size) {
+          if (!isClassBased(lNode, i)) {
+            const styleName = getProp(lNode, i);
+            const value = getValue(lNode, i) as string | null;
+            if (value) {
```

Comment:
```suggestion
            if (value !== null) {
```

We should add a test for e.g. `margin: 0;`. See #27270

Reply:
fixed